lib/awful/widget/taglist.lua

```diff
@@ -91,15 +94,53 @@ function taglist.taglist_label(t, args)
             end
             if bg_empty then bg_color = bg_empty end
             if fg_empty then fg_color = fg_empty end
+
+            if args.shape_empty or theme.taglist_shape_empty then
+                shape = args.shape_empty or theme.taglist_shape_empty
+            end
+
+            if args.shape_border_width_empty or theme.taglist_shape_border_width_empty then
+                shape_border_width = args.shape_border_width_empty or theme.taglist_shape_border_width_empty
+            end
+
+            if args.shape_border_color_empty or theme.taglist_shape_border_color_empty then
+                shape_border_color = args.shape_border_color_empty or theme.taglist_shape_border_color_empty
+            end
         end
     end
     if t.selected then
         bg_color = bg_focus
         fg_color = fg_focus
+
+        if args.shape_focus or theme.taglist_shape_focus then
+            shape = args.shape_focus or theme.taglist_shape_focus
+        end
+
+        if args.shape_border_width_focus or theme.taglist_shape_border_width_focus then
+            shape = args.shape_border_width_focus or theme.taglist_shape_border_width_focus
+        end
+
+        if args.shape_border_color_focus or theme.taglist_shape_border_color_focus then
+            shape = args.shape_border_color_focus or theme.taglist_shape_border_color_focus
+        end
+
     elseif tag.getproperty(t, "urgent") then
         if bg_urgent then bg_color = bg_urgent end
         if fg_urgent then fg_color = fg_urgent end
+
+        if args.shape_urgent or theme.taglist_shape_urgent then
+            shape = args.shape_urgent or theme.taglist_shape_urgent
+        end
+
+        if args.shape_border_width_urgent or theme.taglist_shape_border_width_urgent then
+            shape_border_width = args.shape_border_width_urgent or theme.taglist_shape_border_width_urgent
+        end
+
+        if args.shape_border_color_urgent or theme.taglist_shape_border_color_urgent then
+            shape_border_color = args.shape_border_color_urgent or theme.taglist_shape_border_color_urgent
+        end
```

Comment:
This is documented in a later commit along with some other undocumented things. Mind if it stays that way?